### PR TITLE
BNS: fix & feats

### PR DESCRIPTION
**Pull request by lgalabru**
After having a closer look at https://github.com/blockstack/stacks-blockchain/pull/2277, it appeared to me that:
1) the namespace's price table is wrong
2) the import process is going to import some namespace v1 (BTC based) and silently/implicitly be converting them into namespace v2 (STX based). So #2277 should should probably apply a BTC / STX conversion - I think that could be achieved by tweaking the coefficient in the price function. 

The coefficient update would be arbitrarily fixed by us, set in the stone, and all this without consulting the namespace owners. I'm not super comfortable with this idea and would love to avoid a hard fork because of that :).
This PR is implementing the ability to update the price function + the possibility to forever renounce to this ability.

### CODE REVIEW

**Comment by zone117x** [comment]
Note that the Stacks 1.0 image is blocked on this PR, as the `btc` namespace patch PR is blocked on this one.

File: src/chainstate/stacks/boot/bns.clar
Comment (jcnelson) [inline src/chainstate/stacks/boot/bns.clar:554]: Wrong comment text?

File: bns-test/test/namespace_update_price_function.test.ts
Comment (jcnelson) [inline bns-test/test/namespace_update_price_function.test.ts:166]: Shouldn't this be `should fail`?

**Review by jcnelson — APPROVED** [review]
LGTM!  Please fix the comments before merging.

File: src/chainstate/stacks/boot/bns.clar
Comment (lgalabru) [reply src/chainstate/stacks/boot/bns.clar]: Addressed with https://github.com/blockstack/stacks-blockchain/pull/2286/commits/d9831b806de4414eb16a99c1af714193f6ea9704

File: bns-test/test/namespace_update_price_function.test.ts
Comment (lgalabru) [reply bns-test/test/namespace_update_price_function.test.ts]: Addressed with https://github.com/blockstack/stacks-blockchain/pull/2286/commits/d9831b806de4414eb16a99c1af714193f6ea9704

**Review by kantai — APPROVED** [review]

**Pull request merged**
